README
NAME

    Dist::Zilla::PluginBundle::SHANTANU - Dist Zilla Plugin Bundle the way
    I like to use it

VERSION

    version 0.42

SYNOPSIS

       # in dist.ini
         [@SHANTANU]

DESCRIPTION

    This is a Dist::Zilla PluginBundle. The way I use it. While this bundle
    is customized to my needs, others might be better of forking this
    repository and modifying it to their own needs or using the more mature
    Plugin bundles that this is derived from like the one by David Golden.

ATTRIBUTES

 makemaker

    makemaker attribute By default uses [MakeMaker::Awesome] This can be
    overriden by defining this attribute

 skip_makemaker

    Skip Default Makemaker option to add your own plugin for generating
    makefile

 no_git

    no_git attribute

 no_commitbuild

    no_commitbuild attribute, do not create a build branch

 version_regexp

    version_regexp attribute

 is_task

    Use Taskweaver in lieu of PodWeaver

 weaver_config

    PodWeaver config_plugin attribute

 no_spellcheck

    Skip spelling checks

 exclude_filename

    list of filenames to exclude e.g. exclude_filename=dist.ini
    exclude_filename=META.json exclude_filename=META.yml

 stopwords

    Stopwords to exclude for spell checks in pod

 no_critic

    Skip Perl Critic Checks

 no_coverage

    Skip Pod Coverage tests

 test_kwalitee

    Create kwalitee tests.

 test_compile

    Create compile tests.

 auto_prereq

    Automatically get prerequisites(default 1)

 fake_release

 tag_regexp

    Regex for obtaining the version number from git tag

 compile_for_debian

    generate debian specific files like control etc. Useful if you are
    using dh-make-perl for building .deb files from your package

 tag_format

    Git Tag format

 git_remote

USAGE

    To use this PluginBundle, just add it to your dist.ini. You can provide
    the following options:

      * is_task -- this indicates whether TaskWeaver or PodWeaver should be
      used. Default is 0.

      * auto_prereq -- this indicates whether AutoPrereq should be used or
      not. Default is 1.

      * tag_format -- given to Git::Tag. Default is 'release-%v' to be more
      robust than just the version number when parsing versions for
      Git::NextVersion

      * version_regexp -- given to Git::NextVersion. Default is
      '^release-(.+)$'

      * fake_release -- swaps FakeRelease for UploadToCPAN. Mostly useful
      for testing a dist.ini without risking a real release.

      * weaver_config -- specifies a Pod::Weaver bundle. Defaults to
      @SHANTANU.

      * stopwords -- add stopword for Test::PodSpelling (can be repeated)

      * no_git -- bypass all git-dependent plugins

      * no_critic -- omit Test::Perl::Critic tests

      * no_spellcheck -- omit Test::PodSpelling tests

      * no_coverage -- omit Pod Coverage tests

    When running without git, C<GatherDir> is used instead of
    C<Git::GatherDir>, C<AutoVersion> is used instead of
    C<Git::NextVersion>, and all git check and commit operations are
    disabled.

    This PluginBundle now supports ConfigSlicer, so you can pass in options
    to the plugins used like this:

       [@SHANTANU]
       ExecDir.dir = scripts ; overrides ExecDir

COMMON PATTERNS

 use github instead of RT

       [@SHANTANU]
       :version = 0.32
       AutoMetaResourcesPrefixed.bugtracker.github = user:shantanu
       AutoMetaResourcesPrefixed.bugtracker.rt = 0

SEE ALSO

      * Dist::Zilla

      * Dist::Zilla::Plugin::PodWeaver

      * Dist::Zilla::Plugin::TaskWeaver

SUPPORT

 Bugs / Feature Requests

    Please report any bugs or feature requests through github at
    https://github.com/shantanubhadoria/perl-dist-zilla-pluginbundle-shantanu/issues.
    You will be notified automatically of any progress on your issue.

 Source Code

    This is open source software. The code repository is available for
    public review and contribution under the terms of the license.

    https://github.com/shantanubhadoria/perl-dist-zilla-pluginbundle-shantanu

      git clone git://github.com/shantanubhadoria/perl-dist-zilla-pluginbundle-shantanu.git

AUTHOR

    Shantanu Bhadoria <shantanu@cpan.org> https://www.shantanubhadoria.com

CONTRIBUTORS

      * Karen Etheridge <ether@cpan.org>

      * Shantanu Bhadoria <shantanu att cpan dott org>

COPYRIGHT AND LICENSE

    This software is copyright (c) 2016 by Shantanu Bhadoria.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Changelog for Dist-Zilla-PluginBundle-SHANTANU

0.43 2016-05-30T03:29:13
 - adding badge depot dependencies (Shantanu Bhadoria)

0.42 2016-05-10T04:36:02Z
 - kwalitee options (Shantanu Bhadoria)

0.41 2016-04-07T03:53:30Z
 - travis root (Shantanu Bhadoria)

0.40 2016-04-07T03:32:17Z
 - fixed perl versions for travis; (Shantanu Bhadoria)
 - modified travis (Shantanu Bhadoria)
 - travis update (Shantanu Bhadoria)
 - Updated travis CI file to pre install kwalitee::extra (Shantanu
   Bhadoria)

0.39 2016-04-04T08:53:56Z
 - v0.39 (Shantanu Bhadoria)
 - fixed prereq version (Shantanu Bhadoria)
 - added coverage (Shantanu Bhadoria)
 - Modified reamdme (Shantanu Bhadoria)
 - Added Badges (Shantanu Bhadoria)
 - Adding a Shantanu badge to use my travis repo naming format (Shantanu
   Bhadoria)
 - [ContributorsFromGit] -> [Git::Contributors] (Karen Etheridge)

0.38 2015-05-25T04:26:08Z
 - v0.38 (Shantanu Bhadoria)

0.37 2015-05-25T04:22:08Z
 - v0.37 (Shantanu Bhadoria)
 - Added option to disable compile tests to allow removal of modules
   manually (Shantanu Bhadoria)

0.36 2015-04-08T09:54:58Z
 - v0.36 (Shantanu Bhadoria)
 - Cleanup and fixes (Shantanu Bhadoria)

0.35 2015-04-08T08:39:02Z
 - v0.35 (Shantanu Bhadoria)
 - adding mailmap (Shantanu Bhadoria)

0.34 2015-04-08T08:21:46Z
 - v0.34 (Shantanu Bhadoria)

0.33 2015-04-08T08:10:06Z
 - v0.33 (Shantanu Bhadoria)
 - Travis tuning (Shantanu Bhadoria)

0.32 2015-04-08T07:59:36Z
 - v0.32 (Shantanu Bhadoria)

0.31 2015-04-08T07:47:24Z
 - v0.31 (Shantanu Bhadoria)

0.30 2015-04-08T07:43:17Z
 - v0.30 (Shantanu Bhadoria)

0.29 2015-04-08T07:34:42Z
 - v0.29 (Shantanu Bhadoria)
 - Added more branched to push list (Shantanu Bhadoria)
 - added .travis to master (Shantanu Bhadoria)

0.28 2015-04-08T04:52:44Z
 - v0.28 (Shantanu Bhadoria)
 - Added .travis.yml to dirty list (Shantanu Bhadoria)
 - adding travis to git ignore (Shantanu Bhadoria)
 - Fixed dirty debian files in build Adding the Readmes in git again
   (Shantanu Bhadoria)
 - Trying removal of possibly unnecessary files (Shantanu Bhadoria)
 - local changes (Shantanu Bhadoria)

0.27 2015-03-30T04:57:08Z
 - v0.27 (Shantanu Bhadoria)
 - adding markdown file to dirty list (Shantanu Bhadoria)

0.26 2015-03-23T06:40:12Z
 - v0.26 (Shantanu Bhadoria)
 - dist.ini changes, added author website (Shantanu Bhadoria)

0.25 2015-03-09T10:05:52Z
 - v0.25 (Shantanu Bhadoria)
 - Adding prefixed autometaresources The github repository can now have a
   prefix of perl-* added to the module name. This allows for seperation of
   perl repos from others. (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.24 2014-12-05T06:37:57Z
 - v0.24 (Shantanu Bhadoria)
 - Added new stopwords for pod spellcheck (Shantanu Bhadoria)
 - Fixes to Dist Zilla PluginBundle for exclude_match empty Array
   exceptions (Shantanu Bhadoria)
 - Mode change and minor mods (Shantanu Bhadoria)

0.23 2013-10-16T04:50:45Z
 - v0.23 (Shantanu Bhadoria)
 - Updated to latest Changelogfromgit cpan changes code (Shantanu Bhadoria)

0.22 2013-10-05T03:57:15Z
 - v0.22 (Shantanu Bhadoria)
 - Added debian files (Shantanu Bhadoria)

0.21 2013-09-29T02:45:52Z
 - v0.21 (Shantanu Bhadoria)
 - Moved Constants and Registers lower (Shantanu Bhadoria)

0.20 2013-09-27T10:08:08Z
 - v0.20 (Shantanu Bhadoria)
 - add additional group to Pod weaver (Shantanu Bhadoria)

0.19 2013-09-14T03:38:26Z
 - v0.19 (Shantanu Bhadoria)
 - Added inc and corpus directories to no index list (Shantanu Bhadoria)

0.18 2013-09-12T08:49:01Z
 - v0.18 (Shantanu Bhadoria)
 - v0.16 (Shantanu Bhadoria)

0.17 2013-09-12T08:45:39Z
 - v0.17 (Shantanu Bhadoria)
 - Added README.pod (Shantanu Bhadoria)

0.16 2013-09-12T06:46:08Z
 - v0.16 (Shantanu Bhadoria)
 - Commit Changes test (Shantanu Bhadoria)
 - trying a different Position for Changelog. moveing Changes back to core
   (Shantanu Bhadoria)

0.15 2013-09-12T06:20:47Z
 - Renamed CHANGES to Changes (Shantanu Bhadoria)

0.14 2013-09-12T06:14:58Z
 - Renamed CHANGES to Changes (Shantanu Bhadoria)

0.13 2013-09-12T06:07:57Z
 - Moved to Changelog From git (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.12 2013-09-11T03:21:17Z
 - fixed Deps (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.11 2013-07-11T03:37:54Z
 - v0.11 (Shantanu Bhadoria)
 - Added Perl Version, few fixes to ignore META.yml for git (Shantanu
   Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.10 2013-07-11T01:10:14Z
 - v0.10 (Shantanu Bhadoria)
 - Added Meta YAML generator to help older parsers and KWalitee (Shantanu
   Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.09 2013-07-09T13:54:24Z
 - v0.09 (Shantanu Bhadoria)
 - Modification for exclude_filename bug (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.08 2013-07-09T10:13:16Z
 - v0.08 (Shantanu Bhadoria)

0.07 2013-07-09T10:00:24Z
 - Swapped Dist::Zilla::Plugin::MakeMaker with
   Dist::Zilla::Plugin::MakeMaker::Awesome (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.04 2013-03-05T12:37:47Z
 - v0.04 (Shantanu Bhadoria)
 - Fixed .gitignore to ignore PERLTIDY log (Shantanu Bhadoria)
 - Fixed a spelling mistake (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.03 2013-03-05T12:09:35Z
 - v0.03 (Shantanu Bhadoria)
 - Bump Changes (Shantanu Bhadoria)

0.02 2013-03-05T12:08:16Z
 - v0.02 (Shantanu Bhadoria)
 - Fixed VERSION Comment (Shantanu Bhadoria)

0.01 2013-03-05T10:55:51Z
 - Fixed VERSION Comment (Shantanu Bhadoria)
 - Added Changelogfromgit plugin (Shantanu Bhadoria)
 - fixed dist.ini (Shantanu Bhadoria)
 - Modified gitignore to allow META.yml files added README Pod for githubi
   and CHangelog file (Shantanu Bhadoria)
 - Adding Git ignore file (Shantanu Bhadoria)
 - Initial commit (Shantanu Bhadoria)


lib/Pod/Weaver/PluginBundle/SHANTANU.pm
use strict;
use warnings;

package Pod::Weaver::PluginBundle::SHANTANU;
# VERSION

use Pod::Weaver 3.101638; 
use Pod::Weaver::Config::Assembler;

# Dependencies
use Pod::Weaver::Plugin::WikiDoc ();
use Pod::Elemental::Transformer::List 0.101620 ();
use Pod::Weaver::Section::Badges 0.0402        ();
use Badge::Depot::Plugin::Perl;
use Badge::Depot::Plugin::Kwalitee;
use Badge::Depot::Plugin::Cpantesters;
use Badge::Depot::Plugin::Gratipay;
use Pod::Weaver::Section::Support 1.001        ();
use Pod::Weaver::Section::Contributors 0.001   ();

sub _exp { Pod::Weaver::Config::Assembler->expand_package( $_[0] ) }

my $repo_intro = <<'END';
This is open source software.  The code repository is available for
public review and contribution under the terms of the license.
END

my $bugtracker_content = <<'END';
Please report any bugs or feature requests through github at 
{WEB}.
You will be notified automatically of any progress on your issue.
END

=method mvp_bundle_config

=cut

sub mvp_bundle_config {
    my @plugins;
    push @plugins, (
        [ '@SHANTANU/WikiDoc',  _exp('-WikiDoc'),  {} ],
        [ '@SHANTANU/CorePrep', _exp('@CorePrep'), {} ],
        [ '@SHANTANU/Name',     _exp('Name'),      {} ],
        [ '@SHANTANU/Version',  _exp('Version'),   {} ],
        [ '@SHANTANU/Prelude',     _exp('Region'),  { region_name => 'prelude' } ],
        [ '@SHANTANU/Synopsis',    _exp('Generic'), { header      => 'SYNOPSIS' } ],
        [ '@SHANTANU/Description', _exp('Generic'), { header      => 'DESCRIPTION' } ],
        [ '@SHANTANU/Overview',    _exp('Generic'), { header      => 'OVERVIEW' } ],

        [ '@SHANTANU/Stability', _exp('Generic'), { header => 'STABILITY' } ],
    );
    for my $plugin (
        [ 'Usage',        _exp('Collect'), { command => 'usage' } ],
        [ 'Functions',    _exp('Collect'), { command => 'func' } ],
        [ 'Attributes',   _exp('Collect'), { command => 'attr' } ],
        [ 'Constructors', _exp('Collect'), { command => 'construct' } ],
        [ 'Methods',      _exp('Collect'), { command => 'method' } ],
        [ 'Registers',    _exp('Collect'), { command => 'register' } ],
        [ 'Constants',    _exp('Collect'), { command => 'constant' } ],
      )
    {
        $plugin->[2]{header} = uc $plugin->[0];
        push @plugins, $plugin;
    }

    push @plugins,
      (
        [ '@SHANTANU/Leftovers', _exp('Leftovers'), {} ],
        [ '@SHANTANU/postlude', _exp('Region'), { region_name => 'postlude' } ],
        [
            '@SHANTANU/Badges',
            _exp('Badges'),
            {
                formats        => 'html',
                badge          => ['Perl', 'Shantanutravis', 'Cpantesters', 'Kwalitee', 'Gratipay'],
                -perl_version  => '5.10+',
                -shantanutravis_user   => 'shantanubhadoria',
                -travis_branch => 'build/master',
                -gratipay_user => 'shantanubhadoria',
            }
        ],
        [
            '@SHANTANU/Support',
            _exp('Support'),
            {
                perldoc            => 0,
                websites           => 'none',
                bugs               => 'metadata',
                bugs_content       => $bugtracker_content,
                repository_link    => 'both',
                repository_content => $repo_intro
            }
        ],
        [ '@SHANTANU/Authors', _exp('Authors'), {} ],
        [ '@SHANTANU/Contributors', _exp('Contributors'), {} ],
        [ '@SHANTANU/Legal',   _exp('Legal'),   {} ],
        [ '@SHANTANU/List', _exp('-Transformer'), { 'transformer' => 'List' } ],
      );

    return @plugins;
}


lib/Badge/Depot/Plugin/Shantanutravis.pm
use strict;
use warnings;

package Badge::Depot::Plugin::Shantanutravis;

our $VERSION = '0.42'; # VERSION

# Dependencies
use Moose;
use namespace::autoclean;
use Types::Standard qw/Str HashRef/;
use Path::Tiny;
use JSON::MaybeXS 'decode_json';
with 'Badge::Depot';

# ABSTRACT: Shantanu Bhadoria's Travis plugin for Badge::Depot based off Badge::Depot::Plugin::Travis


has user => (
    is      => 'ro',
    isa     => Str,
    lazy    => 1,
    default => sub {
        my $self = shift;
        if ( $self->has_meta ) {
            return $self->_meta->{'username'}
              if exists $self->_meta->{'username'};
        }
    },
);


has repo => (
    is      => 'ro',
    isa     => Str,
    lazy    => 1,
    default => sub {
        my $self = shift;
        if ( $self->zilla ) {
            return 'perl-' . $self->zilla->name;
        }
    },
);


has branch => (
    is      => 'ro',
    isa     => Str,
    default => 'build/master',
);

has _meta => (
    is        => 'ro',
    isa       => HashRef,
    predicate => 'has_meta',
    builder   => '_build_meta',
);

sub _build_meta {
    my $self = shift;

    if ( $self->zilla ) {
        return {
            repo    => 'perl-' . $self->zilla->name,
            version => $self->zilla->version,
        };
    }

    return {} if !path('META.json')->exists;

    my $json = path('META.json')->slurp_utf8;
    my $data = decode_json($json);

    return {} if !exists $data->{'resources'}{'repository'}{'web'};

    my $repository = $data->{'resources'}{'repository'}{'web'};
    return {}
      if $repository !~
      m{^https://(?:www\.)?github\.com/([^/]+)/(.*)(?:\.git)?$};

    return {
        username => $1,
        repo     => $2,
    };
}


sub BUILD {
    my $self = shift;
    $self->link_url( sprintf 'https://travis-ci.org/%s/%s',
        $self->user, $self->repo );
    $self->image_url( sprintf 'https://api.travis-ci.org/%s/%s.svg?branch=%s',
        $self->user, $self->repo, $self->branch );
    $self->image_alt('Travis status');
}

1;

__END__

=pod

=head1 NAME

Badge::Depot::Plugin::Shantanutravis - Shantanu Bhadoria's Travis plugin for Badge::Depot based off Badge::Depot::Plugin::Travis

=head1 VERSION

version 0.42

=head1 ATTRIBUTES

=head2 user

=head2 repo

=head2 branch

=head1 METHODS

=head2 BUILD

=head1 AUTHOR

Shantanu Bhadoria <shantanu@cpan.org> L<https://www.shantanubhadoria.com>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2016 by Shantanu Bhadoria.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Makefile.PL
# This Makefile.PL for Dist-Zilla-PluginBundle-SHANTANU was generated by
# Dist::Zilla::Plugin::MakeMaker::Awesome 0.38.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

use 5.010;
use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Dist Zilla Plugin Bundle the way I like to use it",
  "AUTHOR" => "Shantanu Bhadoria <shantanu\@cpan.org> L<https://www.shantanubhadoria.com>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-PluginBundle-SHANTANU",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.010",
  "NAME" => "Dist::Zilla::PluginBundle::SHANTANU",
  "PREREQ_PM" => {
    "Badge::Depot" => 0,
    "Badge::Depot::Plugin::Gratipay" => 0,
    "Badge::Depot::Plugin::Travis" => 0,
    "Dist::Zilla" => "4.3",
    "Dist::Zilla::Plugin::AutoMetaResources" => 0,
    "Dist::Zilla::Plugin::ChangelogFromGit::CPAN::Changes" => "0.109",
    "Dist::Zilla::Plugin::ChangelogFromGit::Debian" => 0,
    "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
    "Dist::Zilla::Plugin::CheckExtraTests" => 0,
    "Dist::Zilla::Plugin::CheckMetaResources" => 0,
    "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
    "Dist::Zilla::Plugin::Control::Debian" => 0,
    "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
    "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
    "Dist::Zilla::Plugin::Git::Contributors" => 0,
    "Dist::Zilla::Plugin::Git::NextVersion" => 0,
    "Dist::Zilla::Plugin::InsertCopyright" => 0,
    "Dist::Zilla::Plugin::License" => 0,
    "Dist::Zilla::Plugin::MakeMaker::Awesome" => 0,
    "Dist::Zilla::Plugin::ManifestSkip" => 0,
    "Dist::Zilla::Plugin::MetaJSON" => 0,
    "Dist::Zilla::Plugin::MetaNoIndex" => 0,
    "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
    "Dist::Zilla::Plugin::MetaTests" => 0,
    "Dist::Zilla::Plugin::MetaYAML" => 0,
    "Dist::Zilla::Plugin::MinimumPerl" => 0,
    "Dist::Zilla::Plugin::OurPkgVersion" => 0,
    "Dist::Zilla::Plugin::PerlTidy" => 0,
    "Dist::Zilla::Plugin::PodCoverageTests" => 0,
    "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
    "Dist::Zilla::Plugin::PodWeaver" => 0,
    "Dist::Zilla::Plugin::PruneCruft" => 0,
    "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
    "Dist::Zilla::Plugin::TaskWeaver" => 0,
    "Dist::Zilla::Plugin::Test::Compile" => 0,
    "Dist::Zilla::Plugin::Test::Kwalitee::Extra" => 0,
    "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
    "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
    "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
    "Dist::Zilla::Plugin::Test::Portability" => 0,
    "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
    "Dist::Zilla::Plugin::Test::Version" => 0,
    "Dist::Zilla::Plugin::TravisYML" => 0,
    "Dist::Zilla::PluginBundle::Git" => "2.028",
    "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
    "Dist::Zilla::Role::PluginBundle::Easy" => 0,
    "JSON::MaybeXS" => 0,
    "Moose" => "0.99",
    "Moose::Autobox" => 0,
    "Path::Tiny" => 0,
    "Pod::Elemental::Transformer::List" => "0.101620",
    "Pod::Weaver" => "3.101638",
    "Pod::Weaver::Config::Assembler" => 0,
    "Pod::Weaver::Plugin::WikiDoc" => 0,
    "Pod::Weaver::Section::Badges" => "0.0402",
    "Pod::Weaver::Section::Contributors" => "0.001",
    "Pod::Weaver::Section::Support" => "1.001",
    "Test::Portability::Files" => "0.06",
    "Types::Standard" => 0,
    "autodie" => "2.00",
    "namespace::autoclean" => "0.09",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::Temp" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::More" => 0,
    "blib" => "1.01"
  },
  "VERSION" => "0.42",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Badge::Depot" => 0,
  "Badge::Depot::Plugin::Gratipay" => 0,
  "Badge::Depot::Plugin::Travis" => 0,
  "Dist::Zilla" => "4.3",
  "Dist::Zilla::Plugin::AutoMetaResources" => 0,
  "Dist::Zilla::Plugin::ChangelogFromGit::CPAN::Changes" => "0.109",
  "Dist::Zilla::Plugin::ChangelogFromGit::Debian" => 0,
  "Dist::Zilla::Plugin::CheckChangesHasContent" => 0,
  "Dist::Zilla::Plugin::CheckExtraTests" => 0,
  "Dist::Zilla::Plugin::CheckMetaResources" => 0,
  "Dist::Zilla::Plugin::CheckPrereqsIndexed" => 0,
  "Dist::Zilla::Plugin::Control::Debian" => 0,
  "Dist::Zilla::Plugin::CopyFilesFromBuild" => 0,
  "Dist::Zilla::Plugin::Git::CommitBuild" => 0,
  "Dist::Zilla::Plugin::Git::Contributors" => 0,
  "Dist::Zilla::Plugin::Git::NextVersion" => 0,
  "Dist::Zilla::Plugin::InsertCopyright" => 0,
  "Dist::Zilla::Plugin::License" => 0,
  "Dist::Zilla::Plugin::MakeMaker::Awesome" => 0,
  "Dist::Zilla::Plugin::ManifestSkip" => 0,
  "Dist::Zilla::Plugin::MetaJSON" => 0,
  "Dist::Zilla::Plugin::MetaNoIndex" => 0,
  "Dist::Zilla::Plugin::MetaProvides::Package" => 0,
  "Dist::Zilla::Plugin::MetaTests" => 0,
  "Dist::Zilla::Plugin::MetaYAML" => 0,
  "Dist::Zilla::Plugin::MinimumPerl" => 0,
  "Dist::Zilla::Plugin::OurPkgVersion" => 0,
  "Dist::Zilla::Plugin::PerlTidy" => 0,
  "Dist::Zilla::Plugin::PodCoverageTests" => 0,
  "Dist::Zilla::Plugin::PodSyntaxTests" => 0,
  "Dist::Zilla::Plugin::PodWeaver" => 0,
  "Dist::Zilla::Plugin::PruneCruft" => 0,
  "Dist::Zilla::Plugin::ReadmeAnyFromPod" => 0,
  "Dist::Zilla::Plugin::TaskWeaver" => 0,
  "Dist::Zilla::Plugin::Test::Compile" => 0,
  "Dist::Zilla::Plugin::Test::Kwalitee::Extra" => 0,
  "Dist::Zilla::Plugin::Test::MinimumVersion" => 0,
  "Dist::Zilla::Plugin::Test::Perl::Critic" => 0,
  "Dist::Zilla::Plugin::Test::PodSpelling" => 0,
  "Dist::Zilla::Plugin::Test::Portability" => 0,
  "Dist::Zilla::Plugin::Test::ReportPrereqs" => 0,
  "Dist::Zilla::Plugin::Test::Version" => 0,
  "Dist::Zilla::Plugin::TravisYML" => 0,
  "Dist::Zilla::PluginBundle::Git" => "2.028",
  "Dist::Zilla::Role::PluginBundle::Config::Slicer" => 0,
  "Dist::Zilla::Role::PluginBundle::Easy" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "JSON::MaybeXS" => 0,
  "Moose" => "0.99",
  "Moose::Autobox" => 0,
  "Path::Tiny" => 0,
  "Pod::Elemental::Transformer::List" => "0.101620",
  "Pod::Weaver" => "3.101638",
  "Pod::Weaver::Config::Assembler" => 0,
  "Pod::Weaver::Plugin::WikiDoc" => 0,
  "Pod::Weaver::Section::Badges" => "0.0402",
  "Pod::Weaver::Section::Contributors" => "0.001",
  "Pod::Weaver::Section::Support" => "1.001",
  "Test::More" => 0,
  "Test::Portability::Files" => "0.06",
  "Types::Standard" => 0,
  "autodie" => "2.00",
  "blib" => "1.01",
  "namespace::autoclean" => "0.09",
  "strict" => 0,
  "warnings" => 0
);

unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
